client/components/AdminChallengeEdit.js

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { connect } from 'react-redux';
+
+import { convertBufferToImgSrc } from '../utils';
+
+const _AdminChallengeEdit = props => {
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [difficulty, setDifficulty] = useState(1);
+  const [html, setHtml] = useState('');
+  const [css, setCss] = useState('');
+
+  // TODO upload image file, convert to binary, save in db
+  const [imageWidth, setImageWidth] = useState('');
+  const [imageHeight, setImageHeight] = useState('');
+  const [image, setImage] = useState({ data: '' });
+
+  useEffect(() => {
+    document.getElementById('name').focus();
+  }, []);
+
+  const handleSubmit = ev => {
+    // TODO convert challenge image to binary, then save
+    ev.preventDefault();
+    const challenge = {
+      name,
+      description,
+      difficulty,
+      html,
+      css,
+      imageWidth,
+      imageHeight,
+    };
+
+    axios
+      .post('api/challenges', challenge)
+      .then(() => props.history.push('/admin/challenges'))
+      .catch(err => console.log(err));
```

Comment:
Later on we can set these errors on save, so we can directly see errors from Sequelize

Reply:
Agreed!